node/internal/actors/agent_supervisor.go

```diff
@@ -66,3 +72,23 @@ func (s *AgentSupervisor) queryWorkloads(ctx *goakt.ReceiveContext) {
 	}
 	ctx.Response(&actorproto.WorkloadList{Workloads: workloads})
 }
+
+func (s *AgentSupervisor) setLameDuck(ctx *goakt.ReceiveContext) *actorproto.LameDuckResponse {
+	for _, pid := range ctx.Self().Children() {
+		r, err := ctx.Self().Ask(context.Background(), pid, &actorproto.SetLameDuck{})
+		if err != nil {
+			s.logger.Error("Child actor failed to set lame duck", slog.Any("err", err), slog.String("parent", ctx.Self().ID()), slog.String("child", pid.ID()))
+			return &actorproto.LameDuckResponse{Id: ctx.Self().ID(), Success: false}
+		}
+		resp, ok := r.(*actorproto.LameDuckResponse)
+		if !ok {
+			s.logger.Warn("Unexpected response from agent")
+			return &actorproto.LameDuckResponse{Id: ctx.Self().ID(), Success: false}
+		}
+		if !resp.Success {
+			s.logger.Error("Child workload failed to shutdown", slog.String("child", pid.ID()), slog.Any("err", err))
+			return &actorproto.LameDuckResponse{Id: ctx.Self().ID(), Success: false}
```

Comment:
absolutely never expose an actor's ID anywhere outside the internal actor plumbing.